cmd/agent/subcommands/run/command.go

```diff
@@ -331,6 +332,9 @@ func getSharedFxOption() fx.Option {
 		fx.Provide(func(config config.Component) status.InformationProvider {
 			return status.NewInformationProvider(httpproxyStatus.GetProvider(config))
 		}),
+		fx.Supply(
+			pkgrcclient.WithAgent("core-agent", version.AgentVersion),
```

Comment:
Is best to have concrete types rather than `func(*client.Options)` as things that we supply to Fx

I suggest you create a dedicate `rcclient.Params` struct. Then in the rcclient dependencies declare it as a dependency.

```go
type dependencies struct {
	fx.In

	Log log.Component
    
    Params   types.Params
	Listeners     []types.RCListener          `group:"rCListener"`          // <-- Fill automatically by Fx
	TaskListeners []types.RCAgentTaskListener `group:"rCAgentTaskListener"` // <-- Fill automatically by Fx
}
```

The type Params could look something like:

```go
type Params struct {
	AgentName string
    AgentVersion string
	PollInterval time.Duration
}
```

With that you would be able to change the supply block to something like this: 

```go 
fx.Supply(
  rcclient.Params{
      AgentName: "core-agent",
      AgentVersion: version.AgentVersion,
  }
),
```

From there would be able to set the logic you need for the rclient options